pkg/flow/logging/sink.go

```diff
@@ -95,5 +103,33 @@ func writerSinkLogger(w io.Writer, o SinkOptions) (log.Logger, error) {
 	if o.IncludeTimestamps {
 		l = log.With(l, "ts", log.DefaultTimestampUTC)
 	}
+
 	return l, nil
 }
+
+type fanoutWriter struct {
+	w io.Writer
+	f []loki.LogsReceiver
```

Comment:
As far as I could see in the code, this will get recreated in-place whenever the `logging` block changes and is re-evaluated; I _think_ we don't need a mutex here.